app.go

```diff
@@ -261,22 +263,83 @@ func (t stopTimeoutOption) String() string {
 	return fmt.Sprintf("fx.StopTimeout(%v)", time.Duration(t))
 }
 
+// exitCode set code to exit with on encountering an error when starting or
+// stopping an fx application. This is private until WithLogger is made public.
+func exitCode(c int) Option {
+	return exitCodeOption(c)
+}
+
+type exitCodeOption int
+
+func (e exitCodeOption) apply(app *App) {
+	app.exitCode = int(e)
+}
+
+func (e exitCodeOption) String() string {
+	return fmt.Sprintf("fx.exitCode(%v)", int(e))
+}
```

Comment:
we don't need control of the exit code for the failure path, and we definitely should not have tests that run os.Exit under any circumstances because that exits the test binary. I would rather lose the coverage on the lines with `os.Exit` than have tests silently succeed with `os.Exit(0)`.